logic/temporaryStorage.js

```diff
@@ -0,0 +1,70 @@
+const hashMap = require("hashmap");
+
+class TemporaryStorage {
+
+    constructor(){
+        if(this.pendingTransactions === undefined){
+            this.pendingTransactions = new HashMap();
+        } 
+        if(this.proposedBlocks === undefined){
+            this.proposedBlocks = [];
+        }
+    }
+
+    clearProposedBlocks(){
+        this.proposedBlocks = [];
+    }
+
+    addPendingTransaction(transaction){
+        this.pendingTransactions.set(transaction.id, transaction);
+    }
+
+    addPendingTransactionFromMap(key, value){
+        this.pendingTransactions.set(key, value);
+    }
+
+    addPendingTransactions(transactions){
+        transactions.map((transaction) => {
+            this.addPendingTransaction(transaction);
+        })
+    }
+
+    addProposedBlock(proposedBlock){
+        var exists = false;
+        this.proposedBlocks.map((block) => {
```

Comment:
Map wordt meestal gebruikt om elementen te muteren en direct in een nieuwe array te stoppen, wellicht is een .filter of .find beter en zuiniger met geheugen